javascript/index.js

```diff
@@ -3,38 +3,106 @@
 
 // 🚨🚨🚨 Comment out the below code before you start working on the code
 
-// Out of sync
-  getInstruction("mashedPotatoes", 0, (step1) => {
+// Iteration 1 - using callbacks
+getInstruction("mashedPotatoes", 0, (step0) => {
+  document.querySelector("#mashedPotatoes").innerHTML += `<li>${step0}</li>`;
+  // ... Your code here
+  getInstruction("mashedPotatoes", 1, (step1) => {
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 1, (step2) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 2, (step3) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 3, (step4) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 4, (step5) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
-    document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
-  }, (error) => console.log(error));
 
+    getInstruction("mashedPotatoes", 2, (step2) => {
+      document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
 
+      getInstruction("mashedPotatoes", 3, (step3) => {
+        document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
 
-// Iteration 1 - using callbacks
-// ...
+        getInstruction("mashedPotatoes", 4, (step4) => {
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>Mashed potatoes are ready!</li>`;
+
+          document.getElementById("mashedPotatoesImg").removeAttribute("hidden");
+        });
+      });
+    });
+  });
+});
 
 // Iteration 2 - using promises
-// ...
+obtainInstruction("steak", 0)
+  .then((step0) => {
+    document.querySelector("#steak").innerHTML += `<li>${step0}</li>`;
+    //  ... Your code here
+    return obtainInstruction("steak", 1).then((step1) => {
```

Comment:
Here, instead of chaining these `.then`, if you close the method each time, it get's more readable. 
Otherwise, you are almost doing the syntax for the example above of the 'callback hell'.
Here's a solution for you to review:
```
obtainInstruction('steak', 0)
  .then((step0) => {
    document.querySelector('#steak').innerHTML += `<li>${step0}</li>`;
    return obtainInstruction('steak', 1);
  })
  .then((step1) => {
    document.querySelector('#steak').innerHTML += `<li>${step1}</li>`;
    return obtainInstruction('steak', 2);
  })

(...)

.catch((err) => console.log(err));
```